src/inference/dev_api/openvino/runtime/threading/istreams_executor.hpp

```diff
@@ -185,15 +185,22 @@ class OPENVINO_RUNTIME_API IStreamsExecutor : virtual public ITaskExecutor {
         /**
          * @brief Modify _streams_info_table and related configuration according to user-specified parameters, bind
          * threads to cpu cores if cpu_pinning is true.
-         * @param stream_nums Number of streams specified by user
-         * @param threads_per_stream Number of threads per stream specified by user
+         * @param stream_nums Number of streams specified by user. support value >= 0. If stream_nums = 0, using all
+         * cores of the platform.
+         * @param threads_per_stream Number of threads per stream specified by user. support value >= 0.
+         * If threads_per_stream = 0, calculate threads per stream to use all cores of the platform.
          * @param core_type Cpu type (Big/Little/Any) specified by user
+         * @param enable_hyper_threading whether to use logical cores
+         * @param use_tbb whether to create threads by TBB
          * @param cpu_pinning Whether to bind the threads to cpu cores
+         * @return updated configured values
          */
-        void update_executor_config(int stream_nums,
-                                    int threads_per_stream,
-                                    PreferredCoreType core_type,
-                                    bool cpu_pinning);
+        Config update_executor_config(int stream_nums = 1,
+                                      int threads_per_stream = 0,
+                                      PreferredCoreType core_type = PreferredCoreType::ANY,
+                                      bool enable_hyper_threading = true,
+                                      bool use_tbb = true,
+                                      bool cpu_pinning = false);
```

Comment:
how is it supposed to be used?
I've created `Config my_config` and then decide to update its fields. Why do I need it? 
Can I just create new config `my_config2` and assign it to `my_config = my_config2` ?

Reply:
The main function of `update_executor_config` is generating new parameters `_streams_info_table` and `_cpu_reservation` of configuration which are used in cpu streams executor. If `_streams_info_table` is empty, processing flow must enter legacy logic which will be removed later. Actually, the two parameters can be passed in config, for example: 
`auto executor_config =
            ov::threading::IStreamsExecutor::Config{"Intel GPU plugin executor",
                                                    0,
                                                    0,
                                                    ov::threading::IStreamsExecutor::ThreadBindingType::CORES,
                                                    1,
                                                    0,
                                                    0,
                                                    ov::threading::IStreamsExecutor::Config::PreferredCoreType::BIG,
                                                    {{2, MAIN_CORE_PROC, 1, 0, 0}},
                                                    true};`
But setting `streams_info_table` may be confused to plugin. So we provided this interface to update the new parameters in [PR20472](https://github.com/openvinotoolkit/openvino/pull/20472)

Reply:
what is really confusing is that I create `Config` with `ctor`, and then immediately call `update_executor_config`.

Looks like you are delegating construction of object to another method `update_executor_config`..

Reply:
Ok, no need to call `update_executor_config` now.